app/helpers/application_helper.rb

```diff
@@ -97,9 +97,25 @@ def hover_class(item)
 
   # Check role based authorization for a UI task
   def role_allows?(**options)
-    if options[:feature].nil?
-      $log.debug("Auth failed - no feature was specified (required)")
-      return false
+    features = Array(options[:feature])
+
+    if features.blank?
+       $log.debug("Auth failed - no feature was specified (required)")
+       return false
+     end
+
+    if !Rails.env.production?
+      # Detect if queried features are missing from the database and possibly invalid
+      if MiqProductFeature.where(:identifier => features).count != features.length
+        message = "#{__method__} with non-existing product feature identifier: #{features.inspect} is not supported.  Create it or use an existing one."
+        feature_identifiers = MiqProductFeature.all.pluck(:identifier)
+        if feature_identifiers.length <= 5
+          # TODO: Handle these in a followup. Allow tests that seed "everything" or a few features to get by.
+          # warn "#{message} Note: some features were seeded: #{feature_identifiers.inspect}.  Seed the missing feature(s)."
+        else
+          raise("#{message} Note: detected features: #{feature_identifiers.inspect}")
+        end
+      end
```

Comment:
👀   this is the main part of this PR that now will raise an error in dev/test when the database is seeded with more than a few product features if you try to `role_allows?` on a feature identifier that doesn't existing in the DB.

The commented `warn` is very verbose right now so we might have to tackle that later to find a good way to migrate tests that assume admin / "everything"  to instead seed the features it really needs so that we can verify the feature checks are working correctly.  For now, we don't get verification that the checks are working outside of the ones that do it correctly + the routes_spec.rb